src/main/java/online/talkandtravel/service/impl/ChatServiceImpl.java

```diff
@@ -56,6 +59,31 @@ public class ChatServiceImpl implements ChatService {
   private final MessageMapper messageMapper;
   private final ChatMapper chatMapper;
   private final UserMapper userMapper;
+  private final UserRepository userRepository;
+
+
+  /**
+   * creates private chat between two users
+   * @param dto dto
+   * @return chat id
+   */
+  @Override
+  public Long createPrivateChat(NewPrivateChatDto dto) {
+    Chat privateChat = chatRepository.save(Chat.builder()
```

Comment:
At first, check if every user with provided id exists and fetch that user from db.
For this use private method. Make it similar to existing methods in this class for `getChat` or `getCountry`

Reply:
You can use `getReferenceById()` but you should wrap it in try-catch block, to throw our custom exception for every concrete entity type:
```
private User getUser(Long userId) {
        try {
            return userRepository.getReferenceById(userId);
        } catch (EntityNotFoundException e) {
            throw new UserNotFoundException(userId);
        }
    }